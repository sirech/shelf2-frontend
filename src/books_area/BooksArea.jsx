// @flow

import React from 'react'

import styles from './styles.module.css'

import { Route } from 'react-router-dom'

import BookList from 'book-list'
import Navigation from 'navigation'

import BookCreator from './BookCreator'

const BooksArea = () => (
  <div className={styles.grid}>
    <main>
      <BookCreator />
      <Route exact path="/books/:year" component={BookList} />
    </main>
    <Navigation />
  </div>
)

export default BooksArea
