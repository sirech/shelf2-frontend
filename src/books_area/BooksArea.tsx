// @flow

import React from 'react'

import { Route } from 'react-router-dom'
import styles from './styles.module.css'

import BookCreator from './BookCreator'
import BookList from 'book-list'
import Navigation from 'navigation'

const BooksArea = () => (
  <div className={styles.grid}>
    <main>
      <BookCreator />
      <Route exact path="/books/:year" component={BookList} />
    </main>
    <Navigation />
  </div>
)

export default BooksArea
