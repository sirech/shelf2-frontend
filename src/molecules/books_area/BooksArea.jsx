// @flow

import React from 'react'

import { Route } from 'react-router-dom'
import { Row, Col } from 'reactstrap'

import authenticated from 'organisms/authenticated'
import BookCreator from 'organisms/book_creator'
import BookList from 'organisms/book_list'
import Navigation from 'organisms/navigation'

const AuthBookCreator = authenticated()(BookCreator)

const BooksArea = () => (
  <Row>
    <Col xs='12' sm='8' md='9' lg='10'>
      <AuthBookCreator />
      <Route exact path='/books/:year' component={BookList} />
    </Col>
    <Col sm='4' md='3' lg='2'>
      <Navigation />
    </Col>
  </Row>
)

export default BooksArea
