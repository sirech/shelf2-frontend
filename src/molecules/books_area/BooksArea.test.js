import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import BooksArea from './BooksArea'

describe('components', () => {
  describe('BooksArea', () => {
    it('renders correctly', () => {
      const component = shallow(<BooksArea />)
      expect(toJson(component)).toMatchSnapshot()
    })
  })
})
