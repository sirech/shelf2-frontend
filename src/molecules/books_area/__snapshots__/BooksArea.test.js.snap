// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components BooksArea renders correctly 1`] = `
<Row
  tag="div"
>
  <Col
    lg="10"
    md="9"
    sm="8"
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
      ]
    }
    xs="12"
  >
    <Connect(authenticatedComponent(Connect(BookCreator))) />
    <Route
      component={[Function]}
      exact={true}
      path="/books/:year"
    />
  </Col>
  <Col
    lg="2"
    md="3"
    sm="4"
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
      ]
    }
  >
    <Connect(Navigation) />
  </Col>
</Row>
`;
